### PR TITLE
test list filters refactored and test completed

### PR DESCRIPTION
**Pull request by matthewbrahms**

### CODE REVIEW

**Comment by matthewbrahms** [comment]
need to rework `cache.go` to be able to range over correct new things from the ListFilters function!

File: filters.go [hunk 1]
```diff
@@ -122,19 +100,28 @@ func (client *Client) GetFilter(id ID) (*Filter, error) {
 	return &q.Account.Filter, HandleErrors(err, nil)
 }
 
-func (client *Client) ListFilters() ([]Filter, error) {
+func (client *Client) ListFilters(variables *PayloadVariables) (FilterConnection, error) {
 	var q struct {
 		Account struct {
 			Filters FilterConnection `graphql:"filters"`
```
Comment (rocktavious) [inline filters.go:106]: we are missing the extra arguments on the graphql query here

**Review by rocktavious — CHANGES_REQUESTED** [review]
Missing query definition with variables in use

File: filters.go (hunk 1 above)
Comment (matthewbrahms) [reply filters.go]: done!

File: filters_test.go [hunk 2]
```diff
@@ -50,14 +50,52 @@ func TestGetMissingFilter(t *testing.T) {
 
 func TestListFilters(t *testing.T) {
 	// Arrange
-	client := ATestClient(t, "filter/list")
+	requests := []TestRequest{
+		{`{"query": "query ($after:String!$first:Int!){account{filters{nodes{connective,htmlUrl,id,name,predicates{key,keyData,type,value}},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}",
```
[the remark this answers is not in the document]
Comment (matthewbrahms) [reply filters_test.go]: accounted for them and tested in GraphiQL like you showed me! 👍

File: filters_test.go [hunk 3]
```diff
@@ -50,14 +50,52 @@ func TestGetMissingFilter(t *testing.T) {
...
+									{{ template "filter_kubernetes_response" }}
+								},
+								{
+									{{ template "filter_tier1service_response" }} 
+								}
+							],
+							{{ template "pagination_initial_pageInfo_response" }},
+							"totalCount": 2
+						  }}}}`},
+		{`{"query": "query ($after:String!$first:Int!){account{filters{nodes{connective,htmlUrl,id,name,predicates{key,keyData,type,value}},pageInfo{hasNextPage,hasPreviousPage,startCursor,endCursor},totalCount}}}",
```
[the remark this answers is not in the document]
Comment (matthewbrahms) [reply filters_test.go]: 👍

**Review by rocktavious — APPROVED** [review]
LGTM

**Pull request merged**